components/draggable/index.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal Dependencies
+ */
+import withSafeTimeout from '../higher-order/with-safe-timeout';
+import './style.scss';
+
+const dragImageClass = 'components-draggable__invisible-drag-image';
+const cloneWrapperClass = 'components-draggable__clone';
+const cloneHeightTransformationBreakpoint = 700;
+const clonePadding = 20;
+
+class Draggable extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onDragStart = this.onDragStart.bind( this );
+		this.onDragOver = this.onDragOver.bind( this );
+		this.onDragEnd = this.onDragEnd.bind( this );
+	}
+
+	/**
+	 * Removes the element clone, resets cursor, and removes drag listener.
+	 * @param  {Object} event     The non-custom DragEvent.
+	 */
+	onDragEnd( event ) {
+		const { onDragEnd = noop } = this.props;
+		this.removeDragClone();
+		// Reset cursor.
+		document.body.classList.remove( 'is-dragging-components-draggable' );
+		event.preventDefault();
+
+		this.props.setTimeout( onDragEnd );
+	}
+
+	/*
+	 * Updates positioning of element clone based on mouse movement during dragging.
+	 * @param  {Object} event     The non-custom DragEvent.
+	 */
+	onDragOver( event ) {
+		this.cloneWrapper.style.top =
+			`${ parseInt( this.cloneWrapper.style.top, 10 ) + event.clientY - this.cursorTop }px`;
+		this.cloneWrapper.style.left =
+			`${ parseInt( this.cloneWrapper.style.left, 10 ) + event.clientX - this.cursorLeft }px`;
+
+		// Update cursor coordinates.
+		this.cursorLeft = event.clientX;
+		this.cursorTop = event.clientY;
+	}
+
+	/**
+	 *  - Clones the current element and spawns clone over original element.
+	 *  - Adds a fake temporary drag image to avoid browser defaults.
+	 *  - Sets transfer data.
+	 *  - Adds dragover listener.
+	 * @param  {Object} event					The non-custom DragEvent.
+	 * @param  {string} elementId				The HTML id of the element to be dragged.
+	 * @param  {Object} transferData			The data to be set to the event's dataTransfer - to be accessible in any later drop logic.
+	 */
+	onDragStart( event ) {
+		const { elementId, transferData, onDragStart = noop } = this.props;
+		const element = document.getElementById( elementId );
+
+		// Set a fake drag image to avoid browser defaults. Remove from DOM right after.
+		// event.dataTransfer.setDragImage is not supported yet in IE,
+		// we need to check for its existance first.
```

Comment:
s/existance/existence. Bonus for line reformatting:

```
		// Set a fake drag image to avoid browser defaults. Remove from DOM
		// right after. event.dataTransfer.setDragImage is not supported yet in
		// IE, we need to check for its existance first.
```